src/commands/file-validate/file.validate.ts

```diff
@@ -0,0 +1,141 @@
+import { fsa } from '@chunkd/fs';
+import { CogTiff } from '@cogeotiff/core';
+import { command, number, option, string } from 'cmd-ts';
+import { registerFileSystem } from '../../fs.register.js';
+import { logger } from '../../log.js';
+import { MapSheet, SheetRanges } from '../../utils/mapsheet.js';
+import { config, registerCli, verbose } from '../common.js';
+
+registerFileSystem({ config: 's3://linz-bucket-config/config.json' });
+
+const SHEET_MIN_X = MapSheet.origin.x + 4 * MapSheet.width; // The minimum x coordinate of a valid sheet / tile
+const SHEET_MAX_X = MapSheet.origin.x + 46 * MapSheet.width; // The maximum x coordinate of a valid sheet / tile
+const SHEET_MIN_Y = MapSheet.origin.y - 41 * MapSheet.height; // The minimum y coordinate of a valid sheet / tile
+const SHEET_MAX_Y = MapSheet.origin.y; // The maximum y coordinate of a valid sheet / tile
+
+export class TileIndexException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+export interface FileList {
+  uri: string;
+  tileName: string;
+}
+
+export const commandValidateFiles = command({
+  name: 'file-validate',
```

Comment:
Would be great to name this something similar to the work its doing, which is around checking the tile names against the tiff offset.

`tile-grid-name-validate` `tile-grid-name`